cmd/gentypes/gentypes.go

```diff
@@ -462,13 +458,8 @@ func main() {
 			b.WriteString(emitToplevelType(replaceGoTypename(typeName), typeMap[typeName]))
 			b.WriteString("\n")
 		}
-	}
 
-	// For top-level types, emit implementations of methods for the Message
-	// interface.
-	for _, typeName := range typeNames {
-		typeName = replaceGoTypename(typeName)
-		emitMessageMethods(&b, typeName)
+		emitMethodsForType(&b, typeName)
```

Comment:
I know this technically works for now because it only does anything for ProtocolMessage, but technically it should be called on replaceGoTypename(typeName)

Reply:
Done